controllers/metal3machinetemplate_controller.go

```diff
@@ -113,6 +119,32 @@ func (r *Metal3MachineTemplateReconciler) reconcileNormal(ctx context.Context,
 func (r *Metal3MachineTemplateReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&capm3.Metal3MachineTemplate{}).
+		Watches(
+			&source.Kind{Type: &capm3.Metal3Machine{}},
+			handler.EnqueueRequestsFromMapFunc(r.Metal3MachinesToMetal3MachineTemplate),
+		).
 		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
 		Complete(r)
 }
+
+func (m *Metal3MachineTemplateReconciler) Metal3MachinesToMetal3MachineTemplate(o client.Object) []ctrl.Request {
+	result := []ctrl.Request{}
+	// Fetch the Metal3MachineTemplate instance.
+	metal3MachineTemplate := &capm3.Metal3MachineTemplate{}
+	if m3m, ok := o.(*capm3.Metal3Machine); ok {
+		if m3m.Annotations[clonedFromGroupKind] != "Metal3MachineTemplate.infrastructure.cluster.x-k8s.io" && m3m.Annotations[clonedFromName] != metal3MachineTemplate.Name {
```

Comment:
Isn't this `metal3MachineTemplate.Name` supposed to be empty at this point since it was declared at line 133 and hasn't been initialized since?